.github/workflows/ci.yml

```diff
@@ -0,0 +1,32 @@
+name: CI
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    name: Build
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.18'
+      - name: Check out code
+        uses: actions/checkout@v1
+      - name: cache go mod
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go
+      - name: Build
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.43.0
+          export PATH=$PATH:$(go env GOPATH)/bin
+          make check
```

Comment:
Is it possible to add a simple E2E CI?

Reply:
i will try it

Reply:
https://github.com/containers/nydus-storage-plugin/issues/8

Reply:
<img width="1712" alt="image" src="https://user-images.githubusercontent.com/961094/185844651-77ae706e-da4c-4ec6-a203-1d93f099f1b4.png">

done